src/services/SearchResults.cpp

```diff
@@ -231,15 +231,21 @@ void SearchResults::ProcessBlocksNibbles(const SearchResults& srSource, unsigned
         if (block.GetSize() > vMemory.capacity())
             vMemory.reserve(block.GetSize());
 
-        unsigned char* pMemory = vMemory.data();
-        const unsigned char* pPrev = block.GetBytes();
+        std::vector<unsigned char> vMemory2(vMemory.capacity());
+        _CONSTANT_LOC pPrevSize {
+            sizeof(block.GetBytes()) / sizeof(*block.GetBytes())
+        };
+        std::array<unsigned char, pPrevSize> pPrev{};
```

Comment:
The compiler may not warn with C arrays, but it will warn if `std::array` is uninitialized, it'll fill all the slots with '\0' instead of garbage, leading possible stack corruption.

Reply:
`block.GetBytes()` returns a pointer, not an array. why are you trying to calculate size by dividing the size of the pointer by the size of an item in the pointer? 

pPrev is either going to point to the buffer in the block, or to memory on the heap. It's functionality similar to SSO. If the block of memory is smaller than the size of a pointer, just write it where the pointer would go. I don't even want to try to fathom what this change is trying to do. pPrev is not local data.